lucene/core/src/test/org/apache/lucene/index/TestIndexWriter.java

```diff
@@ -3775,4 +3777,120 @@ public void testRefreshAndRollbackConcurrently() throws Exception {
       IOUtils.close(sm, dir);
     }
   }
+
+  public void testRandomOperations() throws Exception {
+    IndexWriterConfig iwc = newIndexWriterConfig();
+    iwc.setMergePolicy(new FilterMergePolicy(newMergePolicy()) {
+      boolean keepFullyDeletedSegment = random().nextBoolean();
+
+      @Override
+      public boolean keepFullyDeletedSegment(IOSupplier<CodecReader> readerIOSupplier) {
+        return keepFullyDeletedSegment;
+      }
+    });
+    try (Directory dir = newDirectory();
+         IndexWriter writer = new IndexWriter(dir, iwc);
+         SearcherManager sm = new SearcherManager(writer, new SearcherFactory())) {
+      Semaphore permits = new Semaphore(10 + random().nextInt(1000));
```

Comment:
This is the number of operations (across all threads) we will execute?  Maybe rename it to `numOperations` or something?

Reply:
++. Adjusted in https://github.com/apache/lucene-solr/pull/1331/commits/e3fd5ce0643d7a05fd0b530b20b58e7b5589cd2a.